enigma-principal/enclave/src/keys_keeper_t/mod.rs

```diff
@@ -135,16 +138,25 @@ pub(crate) fn ecall_get_enc_state_keys_internal(
     let msg = PrincipalMessage::from_message(msg_bytes)?;
     let user_pubkey = msg.get_pubkey();
     let msg_id = msg.get_id();
-    let sc_addrs: Vec<ContractAddress> = match msg.data {
+    let sc_addrs: Vec<ContractAddress> = match msg.data.clone() {
```

Comment:
You can just move the `image` and the `print` to before this and then you won't need to clone anything...